src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/AndroidX509CertificateReader.cs

```diff
@@ -0,0 +1,492 @@
+            }
+        }
+
+        public byte[] SerialNumber => Interop.AndroidCrypto.X509GetSerialNumber(_cert);
+
+        public string SignatureAlgorithm => Interop.AndroidCrypto.X509GetSignatureAlgorithm(_cert);
+
+        public DateTime NotAfter
+        {
+            get
+            {
+                ulong msFromUnixEpoch = Interop.AndroidCrypto.X509GetNotAfter(_cert);
+                return DateTime.UnixEpoch.AddMilliseconds(msFromUnixEpoch).ToLocalTime();
+            }
+        }
+
+        public DateTime NotBefore
+        {
+            get
+            {
+                ulong msFromUnixEpoch = Interop.AndroidCrypto.X509GetNotBefore(_cert);
+                return DateTime.UnixEpoch.AddMilliseconds(msFromUnixEpoch).ToLocalTime();
+            }
+        }
+
+        public byte[] RawData => Interop.AndroidCrypto.EncodeX509(_cert);
+
+        public int Version
+        {
+            get
+            {
+                int version = Interop.AndroidCrypto.X509GetVersion(_cert);
+                if (version < 0)
+                {
+                    throw new CryptographicException();
+                }
+
+                return version;
+            }
+        }
+
+        public bool Archived
+        {
+            get { return false; }
+            set
+            {
+                throw new PlatformNotSupportedException(
+                    SR.Format(SR.Cryptography_Unix_X509_PropertyNotSettable, nameof(Archived)));
+            }
+        }
+
+        public string FriendlyName
+        {
+            get { return ""; }
+            set
+            {
+                throw new PlatformNotSupportedException(
+                  SR.Format(SR.Cryptography_Unix_X509_PropertyNotSettable, nameof(FriendlyName)));
+            }
+        }
+
+        public X500DistinguishedName SubjectName
+        {
+            get
+            {
+                if (_subjectName == null)
+                {
+                    _subjectName = Interop.AndroidCrypto.X509GetSubjectName(_cert);
+                }
+
+                return _subjectName;
+            }
+        }
+
+        public X500DistinguishedName IssuerName
+        {
+            get
+            {
+                if (_issuerName == null)
+                {
+                    _issuerName = Interop.AndroidCrypto.X509GetIssuerName(_cert);
+                }
+
+                return _issuerName;
+            }
+        }
+
+        [UnmanagedCallersOnly]
+        private static unsafe void EnumExtensionsPolicyDataCallback(byte* oid, int oidLen, byte* data, int dataLen, byte isCritical, void* context)
+        {
+            ref PolicyData policyData = ref Unsafe.As<byte, PolicyData>(ref *(byte*)context);
+            string oidStr = Encoding.UTF8.GetString(oid, oidLen);
+            byte[] rawData = AsnDecoder.ReadOctetString(new ReadOnlySpan<byte>(data, dataLen), AsnEncodingRules.DER, out _);
```

Comment:
This currently allocates for every extension.  It could be made a little more optimal given that `AsnEncodingRules.DER` requires that all octet strings use the primitive encoding.

```C#
try
{
    ReadOnlySpan<byte> rawData;

    if (!AsnDecoder.TryReadPrimitiveOctetString(new ReadOnlySpan<byte>(data, dataLen), AsnEncodingRules.DER, out rawData, out _))
    {
        Debug.Fail("TryReadPrimitiveOctetString can't return false for DER inputs");
        return;
    }

    switch (oidStr)
    {
        case Oids.ApplicationCertPolicies:
            policyData.ApplicationCertPolicies = rawData.ToArray();
            break;
        ...
    }
}
catch (AsnContentException)
{
}
```

The try/catch is, I suppose, optional.  Given that the underlying system has already accepted the certificate there shouldn't be malformed data here.